### PR TITLE
Add yarn 3 to create redwood app template

### PR DESCRIPTION
**Pull request by jtoar**
This PR adds yarn 3 to the create-redwood-app template. This speeds up the `⠏ Running 'yarn install'... (This could take a while)` step significantly.

Note that you can still use yarn 1:

```
yarn create redwood-app --yarn1
```

- [x] what do we want to do with CI? Should the smoke test still be on yarn 1?
  - we'll deploy a few of the projects in the deploy repo using yarn 1 
- [x] yarn 3 has the `create` command too, but we still check for yarn compatibility, so it doesn't work. should we tweak or remove the check for yarn compatibility?
  - change the engine requirements to allow yarn 3
- [x] this works locally and CI seems to be passing, but publishing an experimental release or merging and testing canary are still probably necessary
  - got the ok to merge when ready
- [ ] test project

### CODE REVIEW

**Comment by thedavidprice** [comment]
updated from main including Auth Smoke Tests via #4673

**Comment by thedavidprice** [comment]
@jtoar I started working on changes to build:test-project to update the fixture to use Yarn 3. See https://github.com/redwoodjs/redwood/pull/4761/commits/72992df817e85f5f26dfabf3e3e0cb2d63002f3b

- added new command option `yarn1` that will set version classic (default = false)
- updated yarn install commands to be Yarn 3 compatible

### Rebuilding the Test-project fixture
I think tried (unsuccessfully) to rebuild the `__fixture__/test-project` by running:
```
yarn build:test-project --rebuildFixture
```

> Note: at the very beginning of the process, the console will output the temp directory, which you can copy and open to examine the building process. It will be a timestamp-named dir at a path like `/var/folders/wl/drx1fhwd2pjdmlyypfdt9kqc0000gn/T/redwood-test-project/`

The initial `yarn install` task is successful. But as soon as it tries to run tailwind `yarn workspace web add ...`, yarn fails. I commented out the tailwind steps, which then failed at the following `yarn rw ...` command. The error is always the same if I try the command from the temp test-project directory:
```
➤ YN0000: ┌ Link step
➤ YN0001: │ Error: ENOENT: no such file or directory, lstat '/private/var/folders/wl/drx1fhwd2pjdmlyypfdt9kqc0000gn/T/redwood-test-project/2022-03-16T00-01-49.102Z/node_modules/@redwoodjs/core/node_modules/@babel/cli/node_modules/semver'
➤ YN0000: └ Completed in 0s 683ms
➤ YN0000: Failed with errors in 1s 957ms
```

Perhaps it's now an error running a yarn 3 command inside a Yarn 3 process?

**Comment by jtoar** [comment]
The smoke test's failing on `yarn rw data-migrate up`:

https://github.com/redwoodjs/redwood/runs/5569052576?check_suite_focus=true#step:21:27

File: .github/workflows/smoke-test.yaml [hunk 1]
```diff
@@ -121,9 +121,9 @@ jobs:
           yarn rw prisma migrate dev --name ci-test
         working-directory: ${{ steps.setup_test_project.outputs.test_project_path }}
 
-      - name: Run "prisma generate"
+      - name: Run "prisma migrate reset"
         run: |
-          yarn rw prisma generate
+          yarn rw prisma migrate reset --force
```
[the remark this answers is not in the document]
Comment (jtoar) [reply .github/workflows/smoke-test.yaml]: Didn't work; trying `dm up` before `install`

**Review by thedavidprice — CHANGES_REQUESTED** [review]
@jtoar the code and updates look good. However, I still can't get the build:test-project commands to work.

I just tried `yarn build:test-project --rebuildFixture` and ran into the same error I reported previously with Tailwind and/or any following yarn command.

Did you ever get the commands to work locally?

**Comment by thedavidprice** [comment]
Update here via @jtoar:

> it looks like you can get into a bad install state via yarn rwfw project:copy (on yarn 3 at least) IF you try to install after running it:
> ```
> ➤ YN0000: ┌ Link step
> ➤ YN0001: │ Error: ENOENT: no such file or directory, lstat '/Users/dom/prjcts/rwjs-test-project/node_modules/@redwoodjs/core/node_modules/@babel/cli/node_modules/semver'
> ➤ YN0000: └ Completed in 0s 429ms
> ```
>
> you can fix it by changing the nmMode in .yarnrc.yml to classic but I’m not sure if that’s a viable fix yet or not. tbd!

So this is not specific to the work here. Instead it's something broken with project:copy (but strangely not project:sync)

I'm going to run a local sanity check and then hopefully proceed with this PR as is. Dom will work on project:copy fix in follow-up.

**Comment by thedavidprice** [comment]
> I'm going to run a local sanity check and then hopefully proceed with this PR as is. Dom will work on project:copy fix in follow-up.

I could not get this workaround to, well, work on the build:test-project

**Comment by thedavidprice** [comment]
@jtoar Confirming that create-redwood-app@canary is working great!! 🚀

**Pull request merged**